src/client/clientevent.h

```diff
@@ -73,6 +74,7 @@ struct ClientEventHudChange
 	v2f v2fdata;
 	std::string sdata;
 	u32 data;
+	bool booldata;
```

Comment:
I suggest just putting this into the `u32 data` instead of creating an extra boolean field.

Or actually, you could also have a `textstyle = 0/1/2/3` (0=default,1=bold,2=italic,3=bolditalic) and avoid adding _two_ new properties at the same time.

Reply:
It seemed to me as if it would be cleaner to add a boolvalue field, since putting small things into bigger things is avoided in other cases as well - e.g. the v2fdata could as well be saved into the v3fdata, leaving Z unused. It would be inconsistent to not introduce a new field. Also, why is this not a union in the first place?

Reply:
> e.g. the v2fdata could as well be saved into the v3fdata, leaving Z unused

True, no idea why that wasn't done.

> Also, why is this not a union in the first place? As far as I can see it functions as one.

It does, but the reason is that you cannot put C++ classes into an union.

Reply:
> It does, but the reason is that you cannot put C++ classes into an union.

Not quite -- if it has no constructor or destructor, it's fine, and even constructors and destructors are possible using `new` and a pointer.  So, only `std::string` would need to use `new` for this to be a union.

Reply:
That would definitely make more sense, should it be refactored to work like that?

Reply:
If it's a simple change overall, then probably yes, but otherwise, not in this PR.

Reply:
Irrlicht vectors have constructors, you can't put them in an union.

Reply:
Let me elaborate my statement: C++ 11 allows members of unions to have constructors and destructors, but it won't automatically call destructors for union members since it doesn't track which member is active.  Therefore, if you manually add a destructor to destruct any necessary elements, it's fine.  Irrlicht's vectors have no destructors, and the constructor poses no problem since the correct member will be initialized by other code.

(See, for example, `ClientEvent`.  Although it doesn't use Irrlicht vectors, you could use them just as well as three separate numbers.)

Reply:
Ah that's good to know, thought it was just entirely impossible.

How would this error be fixed though? didn't you say the destructor mattered?
<pre>
whatever.cpp:1204:26: error: call to implicitly-deleted default constructor of 'union (anonymous union at whatever.cpp:1204:2)'
        union { v2f a; v3f b; } test;
                                ^

</pre>

Reply:
Hmm, I didn't realize that it would try to call the constructor.  I've successfully made unions before with Irrlicht vectors.  I think it might be that it tries to initialize the first member but can't since union's can't call constructors or destructors.  If you make the first member e.g. an integer or boolean, I think it would work.  (Also: is it C++11?  Probably, but just making sure.)

Anyhow, unions are a mess in C++, but we unfortunately can't use anything like `std::variant` unless we roll our own.